dspace/modules/api/src/main/java/org/dspace/workflow/WorkflowItem.java

```diff
@@ -283,9 +283,9 @@ public static WorkflowItem findByDOI(Context c, String dataPackageDOI) throws Ap
 
                 if (matched == false) {
                     // count number of authors and number of matched authors: if equal, this is a match.
-                    if (manuscript.authors.author.size() > 0) {
+                    DCValue[] itemAuthors = item.getMetadata("dc", "contributor", "author", Item.ANY);
+                    if (manuscript.authors.author.size() == itemAuthors.length) {
                         int numMatched = 0;
-                        DCValue[] itemAuthors = item.getMetadata("dc", "contributor", "author", Item.ANY);
                         for (int j = 0; j < itemAuthors.length; j++) {
                             for (Author a : manuscript.authors.author) {
                                 double score = JournalUtils.getHamrScore(itemAuthors[j].value, a.fullName());
```

Comment:
Oh! How did I miss that we're already using HAMR in a production class? Mini-celebration time. I wrote the pseudocode for that match function 5 years ago, and it's finally useful.